### PR TITLE
Tests/improve coverage

### PR DESCRIPTION
**Pull request by SchoolGuy**
This aims to push the coverage of the library even further to have a more comprehensive overview of what should work and what not.

### CODE REVIEW

**Comment by SchoolGuy** [comment]
I think I can't push this reasonable further for now. We need to tackle the rest in different PR's. Thus I will redo the Git history and then we can review this in my eyes.

**Comment by SchoolGuy** [comment]
@brejoc Comments welcome!

**Comment by SchoolGuy** [comment]
@agraul Although you are not officially part of the Cobbler Org (yet?). Would you mind doing your first review in here?

File: libcobblersignatures/models/osversion.py [hunk 1]
```diff
@@ -67,124 +67,134 @@ def __eq__(self, other):
...
     @property
-    def kernel_arch(self):
+    def kernel_arch(self) -> str:
         return self._kernel_arch
 
     @kernel_arch.setter
     def kernel_arch(self, value):
-        self._kernel_arch = value
+        if isinstance(value, str):
+            self._kernel_arch = value
```
Comment (brejoc) [inline libcobblersignatures/models/osversion.py:115]: This is silently not setting the value if it's not a string.

File: libcobblersignatures/models/osversion.py [hunk 2]
```diff
@@ -67,124 +67,134 @@ def __eq__(self, other):
...
     @property
-    def version_file_regex(self):
+    def version_file_regex(self) -> str:
         return self._version_file_regex
 
     @version_file_regex.setter
     def version_file_regex(self, value):
-        self._version_file_regex = value
+        if isinstance(value, str):
+            self._version_file_regex = value
```
Comment (brejoc) [inline libcobblersignatures/models/osversion.py:102]: This is silently not setting the value if it's not a string.

File: libcobblersignatures/models/osversion.py [hunk 3]
```diff
@@ -67,124 +67,134 @@ def __eq__(self, other):
...
     @property
-    def supported_arches(self):
+    def supported_arches(self) -> list:
         return self._supported_arches
 
     @supported_arches.setter
     def supported_arches(self, value):
-        self._supported_arches = value
+        if isinstance(value, list):
+            self._supported_arches = value
```
Comment (brejoc) [inline libcobblersignatures/models/osversion.py:141]: This is silently not setting the value if it's not a list.

File: libcobblersignatures/models/osversion.py [hunk 4]
```diff
@@ -67,124 +67,134 @@ def __eq__(self, other):
...
     @property
-    def supported_repo_breeds(self):
+    def supported_repo_breeds(self) -> list:
         return self._supported_repo_breeds
 
     @supported_repo_breeds.setter
     def supported_repo_breeds(self, value):
-        self._supported_repo_breeds = value
+        if isinstance(value, list):
+            self._supported_repo_breeds = value
```
Comment (brejoc) [inline libcobblersignatures/models/osversion.py:154]: This is silently not setting the value if it's not a list.

File: libcobblersignatures/models/osversion.py [hunk 5]
```diff
@@ -67,124 +67,134 @@ def __eq__(self, other):
...
     @property
-    def kernel_file(self):
+    def kernel_file(self) -> list:
         return self._kernel_file
 
     @kernel_file.setter
     def kernel_file(self, value):
-        self._kernel_file = value
+        if isinstance(value, list):
+            self._kernel_file = value
```
Comment (brejoc) [inline libcobblersignatures/models/osversion.py:167]: This is silently not setting the value if it's not a list.

File: libcobblersignatures/models/osversion.py [hunk 6]
```diff
@@ -67,124 +67,134 @@ def __eq__(self, other):
...
     @property
-    def initrd_file(self):
+    def initrd_file(self) -> str:
         return self._initrd_file
 
     @initrd_file.setter
     def initrd_file(self, value):
-        self._initrd_file = value
+        if isinstance(value, str):
+            self._initrd_file = value
```
Comment (brejoc) [inline libcobblersignatures/models/osversion.py:180]: This is silently not setting the value if it's not a string.

File: libcobblersignatures/models/osversion.py [hunk 7]
```diff
@@ -67,124 +67,134 @@ def __eq__(self, other):
...
     @property
-    def isolinux_ok(self):
+    def isolinux_ok(self) -> bool:
         return self._isolinux_ok
 
     @isolinux_ok.setter
     def isolinux_ok(self, value):
-        self._isolinux_ok = value
+        if isinstance(value, bool):
+            self._isolinux_ok = value
```
Comment (brejoc) [inline libcobblersignatures/models/osversion.py:193]: This is silently not setting the value if it's not a boolean.

File: libcobblersignatures/signatures.py [hunk 8]
```diff
@@ -125,10 +118,7 @@ def _importsignaturesfile(self, filepath):
                          by the Python ``open()`` function.
         """
         with open(filepath) as f:
```
Comment (brejoc) [inline libcobblersignatures/signatures.py:120]: Would be better to only open in read mode.

**Review by brejoc — CHANGES_REQUESTED** [review]
There are some occurrences, where there is a type check, but the value is silently not set when the type is not correct. I think that should be changed, otherwise this might lead to nasty surprises in the future.

File: libcobblersignatures/signatures.py (hunk 8 above)
Comment (SchoolGuy) [reply libcobblersignatures/signatures.py]: Done

File: libcobblersignatures/models/osversion.py (hunk 1 above)
Comment (SchoolGuy) [reply libcobblersignatures/models/osversion.py:121]: Now we raise a TypeError()

File: libcobblersignatures/models/osversion.py (hunk 2 above)
Comment (SchoolGuy) [reply libcobblersignatures/models/osversion.py:106]: Now we raise a TypeError()

File: libcobblersignatures/models/osversion.py (hunk 3 above)
Comment (SchoolGuy) [reply libcobblersignatures/models/osversion.py:151]: Now we raise a TypeError()

File: libcobblersignatures/models/osversion.py (hunk 4 above)
Comment (SchoolGuy) [reply libcobblersignatures/models/osversion.py:166]: Now we raise a TypeError()

File: libcobblersignatures/models/osversion.py (hunk 5 above)
Comment (SchoolGuy) [reply libcobblersignatures/models/osversion.py:181]: Now we raise a TypeError()

File: libcobblersignatures/models/osversion.py (hunk 6 above)
Comment (SchoolGuy) [reply libcobblersignatures/models/osversion.py:196]: Now we raise a TypeError()

File: libcobblersignatures/models/osversion.py (hunk 7 above)
Comment (SchoolGuy) [reply libcobblersignatures/models/osversion.py:211]: Now we raise a TypeError()

**Comment by SchoolGuy** [comment]
Noone is complaining. Thus merging.

**Pull request merged**